### PR TITLE
Setup CircleCI

### PR DESCRIPTION
**Pull request by agnieszkajacek**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+                - spree_3_7
+                - spree_4_0
+                - spree_4_1
+                - spree_4_2
+                - spree_master
+              ruby-version:
+                - "3.0"
+                - "2.7"
+                - "2.5"
+                - "2.3"
```
Comment (damianlegawiec) [inline .circleci/config.yml:41]:
```suggestion
                - "3.0"
                - "2.7"
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      - rspec:
+          matrix:
+            parameters:
+              gemfile:
+                - spree_3_1
+                - spree_3_7
+                - spree_4_0
+                - spree_4_1
+                - spree_4_2
+                - spree_master
```
Comment (damianlegawiec) [inline .circleci/config.yml:36]:
```
                - spree_4_3
                - spree_main
```

**Review by damianlegawiec — COMMENTED** [review]
We don't need to test against so many version, upgrade is easy, if someone uses a 3-year old version, it's up to them to patch it themselves

File: .circleci/config.yml (hunk 2 above)
Comment (agnieszkajacek) [reply .circleci/config.yml]: I don't know if I should create `spree_4_3.gemfile` by myself or generate it somehow, but is this file content ok? :point_down: 

```
source "https://rubygems.org"

gem "rails-controller-testing"
gem "rubocop", "~> 0.63.0", require: false
gem "rubocop-rspec", require: false
gem "spree_auth_devise", "~> 4.3.0"
gem "spree_backend", "~> 4.3.0"
gem "spree_core", "~> 4.3.0"
gem "sass-rails"
gem "rspec-rails", "~> 4.0.0.beta2"

gemspec path: "../"

```

File: Appraisals [hunk 3]
```diff
@@ -40,10 +40,19 @@ appraise 'spree-4-2' do
...
+  gem 'spree_auth_devise',        '~> 4.3.0'
+  gem 'rails-controller-testing'
+  gem 'sass-rails'
+  gem 'rspec-rails',              '~> 4.0.0.beta2'
+end
+
+appraise 'spree-main' do
+  gem 'spree_core',               '~> 4.3'
+  gem 'spree_backend',            '~> 4.3'
+  gem 'spree_auth_devise',        '~> 4.4', '>= 4.4.1'
```
Comment (agnieszkajacek) [inline Appraisals:55]: @damianlegawiec There is some problem with these gems, CircleCi fails if we lead it to specific branch:
https://app.circleci.com/pipelines/github/spree-contrib/spree_avatax_official/52/workflows/991215cd-4bc5-4ec8-a677-70b0faf1bed2/jobs/764

https://app.circleci.com/pipelines/github/spree-contrib/spree_avatax_official/53/workflows/934a697b-a24b-4de6-991e-8ef9090a1403/jobs/782

https://app.circleci.com/pipelines/github/spree-contrib/spree_avatax_official/54/workflows/0ee92528-3292-4f36-b284-0805a329f676/jobs/790

but if we set up versions there is another problem: https://app.circleci.com/pipelines/github/spree-contrib/spree_avatax_official/56/workflows/3c68ba09-096e-45c0-9af8-bc420a88f06e/jobs/818


IMO, for spree-main we should not set specific versions, it should take it from branch, I guess.
I consulted with Damian, and we both do not know what's wrong with it and how to fix it.

Could you please take a look at this? :pray:

Comment (damianlegawiec) [reply Appraisals:55]: @agnieszkajacek spree main is spree 4.4, not 4.3

Comment (agnieszkajacek) [reply Appraisals:55]: Yeah, but I set dependencies for spree-main, and 4.3 is sets for them not for spree main 
Not really know what do you mean @damianlegawiec

Comment (agnieszkajacek) [reply Appraisals:55]: But it does not work at all if I set the main branch, CircleCI does not pass

Comment (damianlegawiec) [reply Appraisals:55]: you need to point it to github - `gem 'spree_auth_deivse', github: 'spree/spree_auth_devise', branch: 'main'`

Comment (agnieszkajacek) [reply Appraisals]: maybe I didn't say what I wanted right. From the beginning, I had it set up in the same way as you wrote above. The problem is that when we point to a specific branch, CircleCI shows the errors I mentioned above.

I reverted changes and we point to github and the branch https://github.com/spree-contrib/spree_avatax_official/pull/171/commits/e152d058c3eed937289cb805f6f20fd33f39a528
and as you can see, CircleCI fails:
image

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+              database:
+                - mysql
+                - postgres
+              gemfile:
+                - spree_3_1
+                - spree_3_7
+                - spree_4_0
+                - spree_4_1
+                - spree_4_2
+                - spree_main
```
Comment (damianlegawiec) [inline .circleci/config.yml:82]: @agnieszkajacek let's please drop 3.1, 3.7,4.0 &  4.1. Only 4.2 & 4.3

Comment (agnieszkajacek) [reply .circleci/config.yml]: @damianlegawiec it looks like spree_avatax_official does not work with spree 4.3 :thinking: 
tests fails with: 
```
ActiveRecord::RecordInvalid:
        Validation failed: Stores is invalid
```

it occures with these combinations in both databases:
spree_4_3-3.0
spree_4_3-2.7

can it fail or not?

Comment (agnieszkajacek) [reply .circleci/config.yml]: @damianlegawiec :point_up:

**Pull request merged**